sdk/typespec/typespec_client_core/src/xml.rs

```diff
@@ -14,6 +14,9 @@ use typespec::error::{ErrorKind, Result, ResultExt};
 /// The UTF8 [byte order marker](https://en.wikipedia.org/wiki/Byte_order_mark).
 const UTF8_BOM: [u8; 3] = [0xEF, 0xBB, 0xBF];
 
+/// The XML declaration used when serializing.
+const DECLARATION: &str = r#"<?xml version="1.0" encoding="utf-8"?>\n"#;
```

Comment:
My worry is that we're assuming the content is always UTF-8. While native strings in rust are UTF-8, those `[u8]`s could get converted to another encoding. Do we just ignore this possibility in other languages?

If nothing else, looks like `quick-xml` assumes UTF-8 output anyway: https://docs.rs/quick-xml/latest/src/quick_xml/writer.rs.html#188-190. And, according to its `encoding` feature docs, it only supports u8 code units anyway.

Reply:
While I can't speak for all languages, this is what we've been doing for Go without issue (so far anway).